stubs/caldav/caldav/davclient.pyi

```diff
@@ -0,0 +1,70 @@
+from collections.abc import Iterable, Mapping
+from typing import Any
+from urllib.parse import ParseResult, SplitResult
+
+from requests.auth import AuthBase
+from requests.models import Response
+from requests.structures import CaseInsensitiveDict
+
+from .lib.url import URL
+from .objects import Calendar, DAVObject, Principal
+
+_Element = Any  # actually lxml.etree._Element
+
+class DAVResponse:
+    reason: str
+    tree: _Element | None
+    status: int
+    headers: CaseInsensitiveDict[str]
+    objects: dict[str, dict[str, str]]  # only defined after call to find_objects_and_props()
+    def __init__(self, response: Response) -> None: ...
+    @property
+    def raw(self) -> str: ...
+    def validate_status(self, status: str) -> None: ...
+    def find_objects_and_props(self) -> None: ...
+    def expand_simple_props(
+        self, props: Iterable[Any] = ..., multi_value_props: Iterable[Any] = ..., xpath: str | None = ...
+    ) -> dict[str, dict[str, str]]: ...
+
+class DAVClient:
+    proxy: str | None
+    url: URL
+    headers: dict[str, str]
+    username: str | None
+    password: str | None
+    auth: AuthBase | None
+    ssl_verify_cert: bool | str
+    ssl_cert: str | tuple[str, str] | None
+    def __init__(
+        self,
+        url: str,
+        proxy: str | None = ...,
+        username: str | None = ...,
+        password: str | None = ...,
+        auth: AuthBase | None = ...,
+        ssl_verify_cert: bool | str = ...,
+        ssl_cert: str | tuple[str, str] | None = ...,
+    ) -> None: ...
+    def principal(self, url: str | ParseResult | SplitResult | URL | None) -> Principal: ...
```

Comment:
```suggestion
    def principal(self, *, url: str | ParseResult | SplitResult | URL | None) -> Principal: ...
```
It does `Principal(client=self, *largs, **kwargs)`, which I don't think can actually work when largs is non-empty.

Reply:
I'm not sure what you mean. Can you give an example?

Reply:
The definition of this method has `*largs, **kwargs`, but the only thing it does with its args and kwargs is `Principal(client=self, *largs, **kwargs)`. The signature of Principal is `(client, url=None)`, so the only legal ways to call this method are `.principal()` and `.principal(url=url)`.

So in addition to what I wrote above there should be an `= ...`.